Automattic/jetpack
# My Jetpack: tidy Product card component

This PR iterates over ProductCard components in order to simplify and reduce the code.  There is a bunch of repeated code that can be simplified thanks to the `useProduct()` hook, name conventions, etc. The goal is, again, to make the code more eligible, simpler, but without removing the ability to customize some of them when it's required. `showDeactivate` is an example of it.

<img width="117" alt="image" src="https://user-images.githubusercontent.com/77539/156640190-afdf9ca9-da71-4b96-ac52-db1dd9e148ce.png">

Fixes #

#### Changes proposed in this Pull Request:

* Introduces <ConnectedProductCard/> component.
* SImplifies code in product cards by using <ConnectedProductCard/> instead of <ProductCard/> directly


#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:

* Go to My Jetpack dashboard. Confirm everything works as expected:
  *  Activate/Deactivate Product
  * Add Product link leads to the Interstitial page
  * Confirm it triggers events
  * etc
